contracts/adapters/ParaSwapRepayAdapter.sol

```diff
@@ -180,15 +172,19 @@ contract ParaSwapRepayAdapter is BaseParaSwapBuyAdapter, ReentrancyGuard {
       initiator
     );
 
-    uint256 amountSold =
-      _buyOnParaSwap(
-        buyAllBalanceOffset,
-        paraswapData,
-        collateralAsset,
-        debtAsset,
-        collateralAmount,
-        debtRepayAmount
-      );
+    uint256 amountSold = debtRepayAmount;
+
+    if (collateralAsset != debtAsset) {
+      uint256 amountSold =
```

Comment:
At line 178 there is a shadow declaration of amountSold, and due that is not mutating the declaration of line 175 in the cases where the collateral asset is different from debt asset